rfcs/react-components/components/simple-component-implementation.md

```diff
@@ -0,0 +1,124 @@
+
+#### What makes a component a candidate for this RFC's solution?
+
+- Its implementation is simple.
+- The desired design is achievable adjusting the props of the inner components.
+- It doesn't have complex accessibility requirements.
+- It doesn't have strict design requirements.
+
+## Detailed Design or Proposal
+
+### Option 1: Recipes section in docsite
+
+In this option we will add a section to our docs detailing how to create the component with our already created components.
+
+A great example of a recipe is the [Media Objects recipe](https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Media_objects) by MDN.
+
+#### Folder structure inside docs
+
+```
+\- Concepts
+  |- Introduction
+  |- Developer
+  |- Migration
+  |- Recipes
+    |- Recipe #1
+    |- Recipe #2
+\- Theme
+  |- Color
+  |- Motion and Size
+  |- Typography
+\- Components
+  |- ...
+\- Preview Components
+  |- ...
+```
+
+#### Pros
+
+- No need for a new package in our repo.
+- In the case of a component like persona where its design requires us to change the icon's size, we would provide best practices section that could include a size table for reference.
+
+#### Cons
+
+- Allows for issues where the user-created component doesn't look like the design.
+- Could increase a11y issues in user-created components.
+- Could be overwhelming for the user to follow the recipe.
+
+### Option 2: We provide the component
+
+For this option, we would implement the component and provide it like any other component.
+
+#### Pros
+
+- We can ensure the design follows the design spec.
+- We control the accessibility behavior.
+- Makes it easier for the user.
+
+#### Cons
+
+- Another component package for a simple component.
+- More code to maintain.
+
+### Option 3: Option 1 + Option 2 `@fluentui/react-recipe-hooks`
+
+We provide style hooks such as `useMediaObjectStyles()` or `useMediaObjectClassNames()`. The hook's options would contain what would be the props in a regular component and it would return its classnames so the user can apply them respectively. These hooks could be housed in a package such as `@fluentui/react-recipe-hooks`. A simple example could be:
+
+```jsx
+import { Text } from '@fluentui/react-text';
+import { Image } from '@fluentui/react-image';
+import { useMediaObjectClassNames } from '@fluentui/react-recipe-hooks';
+
+const MyMediaObject = () => {
+  const mediaObjectClassNames = useMediaObjectClassNames({ labelPosition: 'below' });
+
+  return (
+    <div className={mediaObjectClassNames.mediaObject}>
+      <Image
```

Comment:
Design can sometimes have specific requirements on the inner components (i.e. Avatar), how do we handle such cases with a style hook based solution ?

Reply:
This would be an example a part of the recipe we leave up to the user. While option 3 gives a hook, there might be some things we cannot provide, and this could be confusing.